wordpress-mobile/WordPress-FluxC-Android
# Migrate multiple media upload and cancellation tests to mocked network

Converts several live media tests (the ones testing media upload cancellation and multiple concurrent media uploads) to mocked network tests.

This will likely fix #564 since I think all the tests involved will be turned into mocks in the end.

The main reason for doing this is that I noticed our test sites have a huge amount of uploaded media (the WP.com test site ran out of storage recently). I tracked down the cause to these kinds of tests: the cancellation doesn't always go through before the media actually completes, and the uploaded media are left on the server because the tests don't know to delete them.

Making these tests mocked seems like a good solution, as we have other live tests testing actual media upload behavior - the tests I migrated in this PR are mostly testing FluxC behavior and generally make sense as mocked tests.

#### A few extra benefits
* A couple of these tests flake on occasion; possibly due to network hiccups, possibly a concurrency issue. Moving them to the mocked stack should resolve these issues (or make them more common/easier to resolve).
* As the tests were also duplicated between WP.com and XML-RPC; this should speed up the live test runs by a bit.

### TODO:
* [x] The method for giving each mock media upload response a unique remote id could be better; right now it relies on the `RequestQueue` having a thread pool size greater than or equal to 5 (the number of concurrent media uploads in the tests). Probably a `ConcurrentHashMap` or the like prepopulated with unique values should do the trick.
* [x] There is at least one test in `ReleaseStack_UploadTest` also not consistently clearing uploaded media - I'd like to migrate those too.

To test:
1. Run the mocked test suite, make sure it passes
2. Log in to the test sites used in `ReleaseStack_MediaTestWPCom` and`ReleaseStack_MediaTestXMLRPC`
3. Run those two test classes, and make sure no new media were uploaded to the test sites at the end of the tesst run
4. Repeat step 3 for `ReleaseStack_UploadTest` (this one's WPCom only, same test site as `ReleaseStack_MediaTestWPCom`)
